apps/dashboard/app/javascript/xdmod.js

```diff
@@ -239,13 +254,24 @@ function createJobsWidget() {
       console.error(error);
       renderJobs({error: error});
 
-      // error - report back for analytics purposes
-      const analyticsUrl = new URL(analyticsPath('xdmod_jobs_widget_error'), document.location);
-      analyticsUrl.searchParams.append('error', error);
-      fetch(analyticsUrl);
+      reportError('xdmod_jobs_widget_error', error);
     });
 }
 
+function addAnalyticsToJob(jobId) {
+  const analyticsContainer = `#details_${jobId}`;
+  fetch(jobAnalyticsUrl(jobId), { credentials: 'include' })
```

Comment:
Seems like this should be cached - likely for a very long time. I mean, the job's complete - we can likely cache it forever right?

Reply:
Good point. Yes, the data should only be requested once. Will update.

Reply:
Fixed

Reply:
Is it? I don't see where it's being cached.

Reply:
I didn't use a caching method as such. I am using the event listener API to enforce the listener to be executed only once:

https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#once

```
    tr.addEventListener('click', function(event) {
      const jobId = event.currentTarget.getAttribute("data-xdmod-jobid");
      getJobAnalytics(jobId, helpers)
    }, { once: true });
```

Reply:
Can we just use regular fetch cache as well? This would work for 1 page load, but the second time we page load it'd re-request everything all over again.

These jobs are all done, so it's not like the data is ever going to change, no matter how many times you refresh the page.

Reply:
OK - Looking into fetch cache.

Reply:
Standard fetch cache does not work on the jobs analytics API requests as the server sets the no cache header. The browser does not cache the request and the fetch is made to the XDMoD backend.
```
 fetch(jobAnalyticsUrl(jobId, helpers), { credentials: 'include', cache: "force-cache" })
```

I then implemented a solution using `localStorage`:

```
function getJobAnalytics(jobId, helpers) {
  const analyticsContainer = `details_${jobId}`;
  const cacheKey = `xdmod:${helpers.realm}:${jobId}`;
  const cachedData = null; //localStorage.getItem(cacheKey);
  if (cachedData) {
    const data = JSON.parse(cachedData);
    renderJobAnalytics(data, analyticsContainer, helpers);
    return;
  }
```

But reviewing the original solution with a single request per page load and this one using localStorage, I prefer the one using a single request per page load.

If you prefer to use the localStorage cache, I will commit the changes.

Reply:
Thanks for looking into it! The single request per page load will work fine, localStorage seems too complex.